consensus/src/test/kotlin/maru/consensus/qbft/EagerQbftBlockCreatorTest.kt

```diff
@@ -225,14 +227,48 @@ class EagerQbftBlockCreatorTest {
     assertThat(createdBlockBody.executionPayload.transactions).isNotEmpty
   }
 
+  @Test
+  fun `uses latest blockhash when parent block is genesis`() {
+    val genesisExecutionPayload =
+      ethApiClient.eth1Web3j
+        .ethGetBlockByNumber(
+          DefaultBlockParameter.valueOf("earliest"),
+          true,
+        ).send()
+        .block
+        .toDomain()
+    val parentBlock =
+      BeaconBlock(
+        beaconBlockHeader = DataGenerators.randomBeaconBlockHeader(0U), // Genesis block has number 0
+        beaconBlockBody =
+          DataGenerators
+            .randomBeaconBlockBody()
+            .copy(executionPayload = genesisExecutionPayload),
+      )
+    val sealedGenesisBeaconBlock = SealedBeaconBlock(parentBlock, emptySet())
+    val parentHeader = QbftBlockHeaderAdapter(sealedGenesisBeaconBlock.beaconBlock.beaconBlockHeader)
+
+    // Spy on the ExecutionLayerManager to capture calls
+    val spyExecutionLayerManager = Mockito.spy(executionLayerManager)
+    val eagerQbftBlockCreator = setup(sealedGenesisBeaconBlock, parentHeader, spyExecutionLayerManager)
+
+    setNextSecondMillis(0)
+    val acceptedBlockTimestamp = (clock.millis() / 1000)
+    eagerQbftBlockCreator.createBlock(acceptedBlockTimestamp, parentHeader)
+
+    // Verify that getLatestBlockHash() was called since parent block number is 0 (genesis)
+    Mockito.verify(spyExecutionLayerManager).getLatestBlockHash()
```

Comment:
Can we instead verify the parent of the new block?

Reply:
No unfortunately as the parent is same in both cases. Need to verify that the hash it's using the build the block is coming from the executionLayerManager instead of the block executionPayload hash. I've improved the test so that the beacon parent block doesn't have the execution layer hash in it now as well.

Reply:
I think what we really want to verify is that it's able to produce a valid block when `parentHeader` contains an empty Genesis block.

Your test is focused on _how_ rather than _what_, which makes it fragile. In the test shouldn't care _how_ it didn't get stuck on a genesis block, we should just make sure it produces something